### ISSUE TITLE
NHapi.NUnit.ParserTest.TestValidHL7Data failing

### ISSUE THREAD

**Issue by Lumirris** [nHapiNET/nHapi#74]
The following test (see link and code below) fails with the following message:

>   Expected string length 31 but was 39. Strings differ at index 8.
  Expected: "Th\\T\\is\\.br\\is\\.br\\A T\\F\\est\\E\\"
  But was:  "Th\\T\\is\\E\\.br\\E\\is\\E\\.br\\E\\A T\\F\\est\\E\\"

Does anyone have any insight as to why the expected result is what it is in this test? It's the only one that doesn't pass for me after forking this and running it.

[https://github.com/duaneedwards/nHapi/blob/master/NHapi20/NHapi.NUnit/ParserTest.cs#L126](https://github.com/duaneedwards/nHapi/blob/master/NHapi20/NHapi.NUnit/ParserTest.cs#L126)

```
[Test]
public void TestValidHl7Data()
{
	PipeParser parser = new PipeParser();
	ORU_R01 oru = new ORU_R01();
	oru.MSH.MessageType.MessageType.Value = "ORU";
	oru.MSH.MessageType.TriggerEvent.Value = "R01";
	oru.MSH.EncodingCharacters.Value = @"^~\&";
	oru.MSH.VersionID.VersionID.Value = "2.3.1";
	oru.GetPATIENT_RESULT(0).GetORDER_OBSERVATION(0).GetOBSERVATION(0).OBX.ValueType.Value = "FT";
	oru.GetPATIENT_RESULT(0).GetORDER_OBSERVATION(0).OBR.SetIDOBR.Value = "1";
	Varies v = oru.GetPATIENT_RESULT(0).GetORDER_OBSERVATION(0).GetOBSERVATION(0).OBX.GetObservationValue(0);
	ST text = new ST(oru);
	text.Value = @"Th&is\.br\is\.br\A T|est\";
	v.Data = text;
	string encodedData = parser.Encode(oru);

	//Console.WriteLine(encodedData);
	string[] segs = encodedData.Split('\r');
	string[] fields = segs[2].Split('|');
	string data = fields[5];
	Assert.AreEqual(@"Th\T\is\.br\is\.br\A T\F\est\E\", data);
}
```

**Comment by gjdickson** [comment]
I'm having similar problems creating ORU^R01 messages with linebreaks for a project I'm working on (though I am very new to nHapi).

There is note in the code comments in Escape.cs about not all escape sequences being supported and it seems \\.br\ is one of them.  \\.br\ does get included in an escape character Hashtable in the file, but the Hashtable does not actually get used.

**Comment by milkshakeuk** [comment]
This was fixed in #185, which is available in the most recent version [v3.0.0-preview.2](https://github.com/nHapiNET/nHapi/releases/tag/v3.0.0-preview.2).
